### PR TITLE
prevent parent branch infinite loops

### LINKED ISSUE #751: infinite loop when entering self as the parent branch
A very rare case but have done this accidentally a couple times and git-town hangs

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo 

resolves #751

### CODE REVIEW

File: features/shared/entering_parent_branch.feature [hunk 1]
```diff
@@ -71,3 +71,22 @@ Feature: Entering a parent branch name when prompted
     Then Git Town is now aware of this branch hierarchy
       | BRANCH    | PARENT |
       | feature-2 | main   |
+
+
+  Scenario: entering self
+    When I run `git town-sync` and enter "feature-2" and "main"
+    Then I see "Please specify the parent branch of feature-2"
+    And I see "'feature-2' cannot be the parent of itself"
+    Then Git Town is now aware of this branch hierarchy
```
Comment (kevgo) [inline features/shared/entering_parent_branch.feature]: ~~Then~~And

File: features/shared/entering_parent_branch.feature [hunk 2]
```diff
@@ -71,3 +71,22 @@ Feature: Entering a parent branch name when prompted
...
+      | BRANCH    | PARENT |
+      | feature-2 | main   |
+
+
+  Scenario: creating a loop
+    When I run `git town-sync` and enter "feature-1" and "feature-2" and "main"
```
Comment (kevgo) [inline features/shared/entering_parent_branch.feature]: Thoughts on providing these steps via a table?

```cucumber
When I run `git town-sync` and enter:
  | feature-1 |
  | feature-2 |
  | main      |

File: features/shared/entering_parent_branch.feature [hunk 3]
```diff
@@ -71,3 +71,22 @@ Feature: Entering a parent branch name when prompted
...
+    Then I see "Please specify the parent branch of feature-2"
+    And I see "Nested branch loop detected: 'feature-1' is an ancestor 'feature-2'"
+    Then Git Town is now aware of this branch hierarchy
```
Comment (kevgo) [inline features/shared/entering_parent_branch.feature]: ~~Then~~And

File: features/shared/entering_parent_branch.feature [hunk 4]
```diff
@@ -71,3 +71,22 @@ Feature: Entering a parent branch name when prompted
...
+    And I see "'feature-2' cannot be the parent of itself"
+    Then Git Town is now aware of this branch hierarchy
+      | BRANCH    | PARENT |
+      | feature-2 | main   |
+
+
+  Scenario: creating a loop
+    When I run `git town-sync` and enter "feature-1" and "feature-2" and "main"
+    Then I see "Please specify the parent branch of feature-2"
+    And I see "Nested branch loop detected: 'feature-1' is an ancestor 'feature-2'"
```
Comment (kevgo) [inline features/shared/entering_parent_branch.feature]: is an ancestor __of__ feature-2

File: features/shared/entering_parent_branch.feature (hunk 2 above)
Comment (charlierudolph) [inline features/shared/entering_parent_branch.feature]: Sounds good. I'll refactor these in a follow up PR

**Pull request closed without merging**